### PR TITLE
Ai

### PR DESCRIPTION
**Pull request by konnorbeard**
@andrewjmeier updated AI

### CODE REVIEW

File: src/checkers/checkers_game.js [hunk 1]
```diff
@@ -117,4 +115,43 @@ CheckersGame.prototype.playerDidWin = function(player) {
...
+  
+  this.board.tokens.forEach(function(token) {     
+    
+    var tile = this.board.findTileForToken(token);
+    
+    // for each possible destination tile...
+    this.board.tiles.forEach(function(destinationRow) { 
+      destinationRow.forEach(function(destination) { 
+        
+        if (this.isValidMove(token, tile, destination))
```
Comment (andrewjmeier) [inline src/checkers/checkers_game.js]: Should have a `{` at the end of the line

File: tabletop/core/aiplayer.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  var results = [];
+  
+  moves.forEach(function(move) { 
+    
+    var gameCopy = game.copyGameStatus(game);    
+    gameCopy.proposedMove = game.copyMoveForGame(move, gameCopy); 
+    gameCopy.executeMove();
+    results.push({
+      move: move, 
+      score: gameCopy.scoreBoard(game)
```
Comment (andrewjmeier) [inline tabletop/core/aiplayer.js]: scoreBoard takes a player as a parameter and you're passing a game?

File: tabletop/core/component.js [hunk 3]
```diff
@@ -54,9 +54,13 @@ Component.prototype.subscribe = function(callback) {
 */
 Component.prototype.propagate = function(child) {
     var context = this;
-    child.subscribe(function(messageObj) {
+    if (typeof child.subscribe  == 'function') {
```
Comment (andrewjmeier) [inline tabletop/core/component.js]: lets take this stuff out. This was fixed another way

**Comment by andrewjmeier** [comment]
So the AI never looks past the next move?

**Comment by konnorbeard** [comment]
@andrewjmeier no, I tried to do a tree search but couldn't figure it out

**Pull request closed without merging**